### PR TITLE
Feat video listing page

### PR DESCRIPTION
**Pull request by ankit9015**
This PR has following features:

1. Video database added
2. Video Listing Page
3. SPA for video
4. Banner is linked with their respective video

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -39,13 +41,17 @@ html {
...
 .main {
   grid-row: 2 / 3;
+  overflow-y: scroll;
+  height: 100%;
+  background-color: var(--black2-bg);
 }
 
 .sidebar {
   grid-row: 2 / 3;
   z-index: 99;
```
Comment (Kothalkarpriya) [inline src/App.css:51]: I don't find it good to have this much z-index. If you can do having 2 as a z-index why use 99 here.

File: src/components/Banner/Banner.css [hunk 2]
```diff
@@ -54,7 +54,7 @@ img.banner__img {
   position: absolute;
 
   top: 80%;
-  z-index: 99;
+  z-index: 2;
```
Comment (Kothalkarpriya) [inline src/components/Banner/Banner.css:57]: Like this

File: src/components/VideoCard/VideoCard.css [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  border-radius: 10px;
+}
+
+.video-card .card__media {
+  border-radius: 0;
+}
+
+.video-card__buttons {
+  display: flex;
+  gap: 20px;
```
Comment (Kothalkarpriya) [inline src/components/VideoCard/VideoCard.css:50]: use rem for more than 10px , also earlier u have did same

File: src/context/VideoContext/VideoContext.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  const [videos, setVideos] = useState([]);
+  const [categories, setCategories] = useState([]);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const response = await axios.get("/api/videos");
+        setVideos(response.data.videos);
+      } catch (err) {
+        console.log(err);
```
Comment (Kothalkarpriya) [inline src/context/VideoContext/VideoContext.js:16]: Good that you used error handling

File: src/pages/Watch/Watch.css [hunk 5]
```diff
@@ -0,0 +1,59 @@
+.watch {
+  display: flex;
+  gap: 15px;
```
Comment (Kothalkarpriya) [inline src/pages/Watch/Watch.css:3]: here also use rem

File: src/pages/Watch/Watch.css [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+.watch__video-container {
+  width: 70%;
+  background-color: var(--dark-black);
+  flex-grow: 1;
+  height: max-content;
+}
+
+.video-container__buttons {
+  gap: 20px;
+  margin: 20px;
```
Comment (Kothalkarpriya) [inline src/pages/Watch/Watch.css]: here also

**Review by Kothalkarpriya — APPROVED** [review]
The Code is clean and understandable, added some suggestions for improvement. Do look into it once. After that, you can merge

File: src/components/VideoCard/VideoCard.css (hunk 3 above)
Comment (ankit9015) [reply src/components/VideoCard/VideoCard.css:50]: This is to make sure that there is a minimum 20px distance between clickables at every viewport width.

**Review by SaurabhChawda — APPROVED** [review]
Liked Your design and Code Looks good to me.

**Pull request merged**